src/mul.cpp

```diff
@@ -413,9 +413,13 @@ RCP<const Basic> mul_expand_two(const RCP<const Basic> &a, const RCP<const Basic
                 // The main bottleneck here is the mul(p.first, q.first) command
                 RCP<const Basic> term = mul(p.first, q.first);
                 if (is_a_Number(*term)) {
-                    iaddnum(outArg(coef), rcp_static_cast<const Number>(term));
+                    iaddnum(outArg(coef),
+                        mulnum(mulnum(p.second, q.second), rcp_static_cast<const Number>(term)));
                 } else {
-                    Add::dict_add_term(d, mulnum(p.second, q.second), term);
```

Comment:
Here we are assuming that `term` does not contain any numbers, but if `p.first = q.first = sqrt(3)`, then `term = 3`. Therefore `(1+sqrt(2)*x)*(1+sqrt(2)*y)` is given as `1 + sqrt(2)*x + sqrt(2)*y + 4*x*y` where `4*x*y , 1` is a pair in the dict of `Add`